lib/srv/db/cloud/iam.go

```diff
@@ -217,6 +217,10 @@ func (c *IAM) getPolicyName() (string, error) {
 func (c *IAM) processTask(ctx context.Context, task iamTask) error {
 	configurator, err := c.getAWSConfigurator(ctx, task.database)
 	if err != nil {
+		if trace.Unwrap(err) == credentials.ErrNoValidProvidersFoundInChain {
+			c.log.Debug("No credentials provider. Skipping IAM task for database %v.", task.database.GetName())
```

Comment:
I Wonder if we should switch to WARN log severity. The `processTask` is triggered only for `RDS` or `Redshift` databases so in case RDS or Redshift  db in database service config or auto dissevered databases  we will silently skip the IMA configuration and add this database as teleport resource. 

Also can we add AWS keyword to the  -> "No AWS credentials provider" string ?

Reply:
When there are no AWS credentials, databases cannot be auto-discovered in the first place. So this would only happen for manually configured RDS or Redshift. That being said, in that case, a WARN is more appropriate. Will update and add the "AWS" part.